### PR TITLE
Enum parsing errors

### PR DESCRIPTION
**Pull request by ohinds**

### CODE REVIEW

File: recidiviz/ingest/scrape/regions/us_ky_hart/us_ky_hart_scraper.py [hunk 1]
```diff
@@ -42,10 +42,11 @@ def get_jailtracker_index(self):
     def get_enum_overrides(self):
         overrides_builder = EnumOverrides.Builder()
 
+        overrides_builder.add('179 DAYS OR PAY', BondType.CASH)
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_ky_hart/us_ky_hart_scraper.py]: `endswith('DAYS OR PAY')`?

Comment (ohinds) [reply recidiviz/ingest/scrape/regions/us_ky_hart/us_ky_hart_scraper.py]: Done

File: recidiviz/common/constants/charge.py [hunk 2]
```diff
@@ -123,34 +124,50 @@ def _get_default_map():
...
     'CONVICTED': ChargeStatus.CONVICTED,
     'COUNTY JAIL TIME': ChargeStatus.SENTENCED,
     'DECLINED TO PROSECUTE': ChargeStatus.DROPPED,
+    'DISMISS': ChargeStatus.DROPPED,
     'DISMISSAL': ChargeStatus.DROPPED,
     'DISMISSED': ChargeStatus.DROPPED,
     'DISMISSED AT COURT': ChargeStatus.DROPPED,
     'DISMISSED BY DISTRICT ATTORNEY': ChargeStatus.DROPPED,
     'DROPPED': ChargeStatus.DROPPED,
+    'ENTERED IN ERROR': None,
```
Comment (jamwalla) [inline recidiviz/common/constants/charge.py]: dropped or acquitted?

File: recidiviz/common/constants/charge.py [hunk 3]
```diff
@@ -198,11 +223,18 @@ def _get_default_map():
     'CRIMINAL': CourtType.DISTRICT,
     'DISTRICT': CourtType.DISTRICT,
     'DISTRICT COURT': CourtType.DISTRICT,
+    'FAMILY': CourtType.OTHER,
```
Comment (jamwalla) [inline recidiviz/common/constants/charge.py]: is this civil court?

File: recidiviz/ingest/scrape/regions/us_ky_boyle/us_ky_boyle_scraper.py [hunk 4]
```diff
@@ -40,7 +42,9 @@ def get_jailtracker_index(self):
     def get_enum_overrides(self) -> EnumOverrides:
         overrides_builder = super(UsKyBoyleScraper,
                                   self).get_enum_overrides().to_builder()
-        overrides_builder.ignore('BONDED', BondType)
         overrides_builder.ignore('10', BondType)
+        overrides_builder.ignore('BONDED', BondType)
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_ky_boyle/us_ky_boyle_scraper.py]: should everything in this file go in the global map?

File: recidiviz/ingest/scrape/regions/us_ky_bullitt/us_ky_bullitt_scraper.py [hunk 5]
```diff
@@ -40,5 +41,6 @@ def get_jailtracker_index(self):
     def get_enum_overrides(self) -> EnumOverrides:
         overrides_builder = super(UsKyBullittScraper,
                                   self).get_enum_overrides().to_builder()
+        overrides_builder.add('RELEASED', ChargeStatus.PRETRIAL)
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_ky_bullitt/us_ky_bullitt_scraper.py]: ?

File: recidiviz/ingest/scrape/regions/us_ky_clark/us_ky_clark_scraper.py [hunk 6]
```diff
@@ -40,4 +41,5 @@ def get_jailtracker_index(self):
     def get_enum_overrides(self) -> EnumOverrides:
         overrides_builder = super(UsKyClarkScraper,
                                   self).get_enum_overrides().to_builder()
+        overrides_builder.add('NOTICE OF DISCHARGE', ChargeStatus.DROPPED)
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_ky_clark/us_ky_clark_scraper.py]: global?

File: recidiviz/ingest/scrape/regions/us_nc_cleveland/us_nc_cleveland_scraper.py [hunk 7]
```diff
@@ -24,3 +31,23 @@ class UsNcClevelandScraper(SuperionScraper):
...
+                      ingest_info: IngestInfo) -> Optional[ScrapedData]:
+        scraped_data: Optional[ScrapedData] = super(
+            UsNcClevelandScraper, self).populate_data(content, task,
+                                                      ingest_info)
+        ingest_info = scraped_data.ingest_info if scraped_data else ingest_info
+
+        for charge in ingest_info.get_all_charges():
+            if charge.bond and charge.bond.bond_type and \
+               charge.bond.bond_type == 'OUT ON BOND':
+                charge.release_reason = ReleaseReason.BOND
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_nc_cleveland/us_nc_cleveland_scraper.py]: `charge.release_reason` should be a string - I would set it to `bond.bond_type` and then make sure it's mapped the right way there.

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py [hunk 8]
```diff
@@ -518,14 +520,49 @@ def _any_charge_has_bond_amount(self, charges):
...
-                    charge.status = charge.bond.bond_type
-                charge.bond.bond_type = None
+            if charge.status:
+                if charge.status == 'UNSECURED_BOND':
+                    charge.bond.bond_type = charge.status
+                    charge.status = ChargeStatus.PRETRIAL.value
+                elif 'FEDERAL' in charge.status:
+                    booking.create_hold(jurisdiction_name=charge.status)
+                    charge.status = None
+                elif 'SENTENCED' in charge.status:
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: this looks like an overrides type of thing? (lambda endswith)

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py [hunk 9]
```diff
@@ -518,14 +520,49 @@ def _any_charge_has_bond_amount(self, charges):
...
+                    charge.status = ChargeStatus.SENTENCED
+
+            if charge.court_type and charge.court_type in {
+                    'OTHER_COUNTY',
+                    'OUT OF STATE TRANSIT'}:
+                booking.create_hold(charge.court_type)
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: create_hold should only take kwargs right? maybe this is `jurisdiction_name`

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py [hunk 10]
```diff
@@ -518,14 +520,49 @@ def _any_charge_has_bond_amount(self, charges):
...
+                    charge.bond.bond_type = None
+                elif charge.bond.bond_type == 'POSTED BOND':
+                    if not charge.status:
+                        charge.status = ChargeStatus.PRETRIAL.value
+                    charge.bond.bond_type = None
+                elif 'HOLD' in charge.bond.bond_type:
+                    booking.create_hold(charge.bond.bond_type)
+                    charge.bond.bond_type = None
+
+            if charge.bond and charge.bond.bond_type:
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: interesting - just to be sure, we're only pointing charges *with bonds* to the blanket bond? Also (not in this PR) is this something the converter should do?

File: recidiviz/common/constants/charge.py (hunk 2 above)
Comment (ohinds) [reply recidiviz/common/constants/charge.py]: I think this means it wasn't ever really even a charge? So I'm choosing to ignore it then.

File: recidiviz/common/constants/charge.py (hunk 3 above)
Comment (ohinds) [reply recidiviz/common/constants/charge.py]: Oh yeah good point.

File: recidiviz/ingest/scrape/regions/us_ky_boyle/us_ky_boyle_scraper.py (hunk 4 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/regions/us_ky_boyle/us_ky_boyle_scraper.py]: I moved some, left others that seemed more specific.

File: recidiviz/ingest/scrape/regions/us_ky_bullitt/us_ky_bullitt_scraper.py (hunk 5 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/regions/us_ky_bullitt/us_ky_bullitt_scraper.py]: Went back and checked on this error. The record for which this was originally seen isn't there anymore, but one with a BondType RELEASED is. I'm going to remove the charge status mapping and just map based on the bond type, so we'll fail later and I can make a new mapping based on that one.

File: recidiviz/ingest/scrape/regions/us_ky_clark/us_ky_clark_scraper.py (hunk 6 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/regions/us_ky_clark/us_ky_clark_scraper.py]: Done.

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py (hunk 8 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: Done.

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py (hunk 9 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: Done

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py (hunk 10 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: Yup! The other ones seem to say 'INCLUDED' and one says 'BLANKET'. We could do it in the converter eventually, yes.

File: recidiviz/ingest/scrape/regions/us_nc_cleveland/us_nc_cleveland_scraper.py (hunk 7 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/regions/us_nc_cleveland/us_nc_cleveland_scraper.py]: Doneish!

**Pull request merged**